pkg/controllers/machine_controller.go

```diff
@@ -554,6 +561,21 @@ func (r *MachineReconciler) domainFor(
 	return domainDesc, volumeStates, nicStates, nil
 }
 
+func (r *MachineReconciler) setDomainMetadata(machine *api.Machine, domain *libvirtxml.Domain) error {
+	domainMetadata := &providermeta.LibvirtProviderMetadata{
+		Namespace: machine.Metadata.Labels[ironcorev1alpha1.MachineNamespaceLabel],
+		Name:      machine.Metadata.Labels[ironcorev1alpha1.MachineNameLabel],
+	}
+	domainMetadataXML, err := xml.Marshal(domainMetadata)
+	if err != nil {
+		return err
+	}
+	domain.Metadata = &libvirtxml.DomainMetadata{
+		XML: string(domainMetadataXML),
+	}
+	return nil
+}
```

Comment:
Can we here maybe encode the entire labels we receive from `machine.Metadata.Labels`? 
I think a generic xml encoder for map[string]string would be nice in order to store everything (indeed from the label name)

Reply:
At SAP we need this information since it contains the top most cluster info as well as the info from the poollet.
It looks like this: 
```
          "downward-api.machinepoollet.api.onmetal.de/root-machine-name": "machine",
          "downward-api.machinepoollet.api.onmetal.de/root-machine-namespace": "default",
          "downward-api.machinepoollet.api.onmetal.de/root-machine-uid": "0276f002-a4d4-4b84-a1b4-94dfaa5475bb",
          "machinepoollet.api.onmetal.de/machine-name": "749112270492dadb02e5313b447429dcc6b56b72ce072f112a44961e49c61eb",
          "machinepoollet.api.onmetal.de/machine-namespace": "machinepoollet-system",
          "machinepoollet.api.onmetal.de/machine-uid": "151e0229-f3c3-456c-bf32-65d67dfb6096"
```